tests/test_grammar.py

```diff
@@ -111,17 +113,17 @@
     ("str_ws_1", "hello world", "hello world"),
     ("str_ws_2", "a b\tc  \t\t  d", "a b\tc  \t\t  d"),
     ("str_esc_ws_1", r"\ hello\ world\ ", " hello world "),
-    ("str_esc_ws_2", "\\ \\\t\\\t", " \t\t"),
+    ("str_esc_ws_2", fr"\ \{TAB}\{TAB}", f" {TAB}{TAB}"),
     ("str_esc_comma", r"hello\, world", "hello, world"),
     ("str_esc_colon", r"a\:b", "a:b"),
     ("str_esc_equal", r"a\=b", "a=b"),
     ("str_esc_parentheses", r"\(foo\)", "(foo)"),
     ("str_esc_brackets", r"\[foo\]", "[foo]"),
     ("str_esc_braces", r"\{foo\}", "{foo}"),
-    ("str_esc_backslash", r"\\", "\\"),
+    ("str_esc_backslash", r" \ ".strip(), r" \ ".strip()),
```

Comment:
put a comment explaining the strip hack, it's not obvious what it is and why you do it.

Reply:
Done in 6015a71